swift/Tests/Blockchains/OasisTests.swift

```diff
@@ -0,0 +1,42 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+import XCTest
+import WalletCore
+
+class OasisTests: XCTestCase {
+
+    func testAddress() {
+
+        let key = PrivateKey(data: Data(hexString: "4f8b5676990b00e23d9904a92deb8d8f428ff289c8939926358f1d20537c21a0")!)!
+        let pubkey = key.getPublicKeyEd25519()
+        let address = AnyAddress(publicKey: pubkey, coin: .oasis)
+        let addressFromString = AnyAddress(string: "oasis1qzawzy5kaa2xgphenf3r0f5enpr3mx5dps559yxm", coin: .oasis)!
+
+        XCTAssertEqual(pubkey.data.hexString, "93d8f8a455f50527976a8aa87ebde38d5606efa86cb985d3fb466aff37000e3b")
+        XCTAssertEqual(address.description, addressFromString.description)
+    }
+
+    func testSign() {
+        let privateKey = PrivateKey(data: Data(hexString: "0x4f8b5676990b00e23d9904a92deb8d8f428ff289c8939926358f1d20537c21a0")!)!
+        let input = OasisSigningInput.with {
+            $0.privateKey = privateKey.data
+            $0.transfer = OasisTransferMessage.with {
+                $0.to = "oasis1qrrnesqpgc6rfy2m50eew5d7klqfqk69avhv4ak5"
+                $0.method = "staking.Transfer"
```

Comment:
```suggestion
```

Reply:
I cannot remove this line. It is required to the test to work.

Reply:
just set it in cpp code, for `OasisTransferMessage`

Reply:
Do you want me to fix this value in the CBOR encoding function and remove method as part of the proto file?

Reply:
yes, probably when building `Transaction` class, not inside CBOR encoding func,

Reply:
Done too!